Classes/RedirectService.php

```diff
@@ -80,33 +89,30 @@ public function buildResponseIfApplicable(Request $httpRequest): ?Response
     }
 
     /**
-     * @param Request $httpRequest
+     * @param ServerRequestInterface $httpRequest
      * @param RedirectInterface $redirect
-     * @return Response|null
+     * @return ResponseInterface|null
      * @throws Exception
      */
-    protected function buildResponse(Request $httpRequest, RedirectInterface $redirect): ?Response
+    protected function buildResponse(ServerRequestInterface $httpRequest, RedirectInterface $redirect): ?ResponseInterface
     {
         if (headers_sent() === true && FLOW_SAPITYPE !== 'CLI') {
             return null;
         }
 
-        $response = new Response();
         $statusCode = $redirect->getStatusCode();
-        $response->setStatus($statusCode);
+        $response = $this->responseFactory->createResponse($statusCode);
 
         if ($statusCode >= 300 && $statusCode <= 399) {
             $location = $redirect->getTargetUriPath();
 
             if (parse_url($location, PHP_URL_SCHEME) === null) {
-                $location = $httpRequest->getBaseUri() . $location;
+                $location = $httpRequest->getAttribute(ServerRequestAttributes::BASE_URI) . $location;
```

Comment:
So this works fine:

```suggestion
                 $location = '' . $httpRequest->getUri()->withQuery('')->withFragment('')->withPath($location);
```

Reply:
well, `->getUri()->...` returns an `UriInterface`, but below the header is expected to be a string. So an explicit cast would probably be better here.
```suggestion
                $location = (string)$httpRequest->getUri()->withQuery('')->withFragment('')->withPath($location);
```